broker-core/src/test/java/io/zeebe/broker/workflow/processor/DeploymentCreatedProcessorTest.java

```diff
@@ -110,6 +113,102 @@ public void shouldNotWriteCloseSubscriptionIfNotMessageStart() {
         .isEqualTo(MessageStartEventSubscriptionIntent.OPEN);
   }
 
+  @Test
+  public void shouldCloseSubscriptionWhenInCorrectOrder() {
+    // given
+    streamProcessor.start();
+
+    // when
+    writeMessageStartRecord(3, 1);
+    writeNoneStartRecord(7, 2);
+
+    // then
+    waitUntil(() -> rule.events().onlyMessageStartEventSubscriptionRecords().count() == 2);
+
+    final TypedRecord<MessageStartEventSubscriptionRecord> closeRecord =
+        rule.events()
+            .onlyMessageStartEventSubscriptionRecords()
+            .withIntent(MessageStartEventSubscriptionIntent.CLOSE)
+            .getFirst();
+
+    assertThat(closeRecord.getValue().getWorkflowKey()).isEqualTo(3);
+  }
+
+  @Test
+  public void shouldIgnoreOutdatedDeployment() {
+    // given
+    streamProcessor.start();
+
+    // when
+    streamProcessor.blockAfterMessageStartEventSubscriptionRecord(
+        r ->
+            r.getValue().getWorkflowKey() == 5
+                && r.getMetadata().getIntent() == MessageStartEventSubscriptionIntent.OPEN);
+    writeMessageStartRecord(5, 2);
+    waitUntil(() -> streamProcessor.isBlocked());
+
+    streamProcessor.blockAfterDeploymentEvent(
+        r -> r.getKey() == 3 && r.getMetadata().getIntent() == DeploymentIntent.CREATED);
+    writeMessageStartRecord(3, 1);
```

Comment:
Can we also have a test which uses here a  normal deployment without message start event, would like to see how it behaves then.

Reply:
Sure, what do you want to see in that test? The current tests already contain a none start event deployment so we can that it doesn't open subscriptions but it does close them.